compiler/src/dmd/expressionsem.d

```diff
@@ -16180,7 +16180,9 @@ private Expression toLvalueImpl(Expression _this, Scope* sc, const(char)* action
         // convert (econd ? e1 : e2) to *(econd ? &e1 : &e2)
         CondExp e = cast(CondExp)(_this.copy());
         e.e1 = _this.e1.toLvalue(sc, action).addressOf();
+        checkAddressable(e.e1, sc);
         e.e2 = _this.e2.toLvalue(sc, action).addressOf();
+        checkAddressable(e.e2, sc);
```

Comment:
I suspect this is just painting lipstick on the pig, if it's not addressable, then it should fail the `toLvalue` conversion done before this check.

Reply:
Well, conditional expressions are not lvalues in C (6.5.15 [note 110] A conditional expression does not yield an lvalue).

As for lvalue conditional expressions in D, I think we should fix the rewrite to not use `*(c ? &a : &b)`

Depending on whether `val` has a side effect, either
```d
(c ? a = val : b = val);
// or
auto tmp = val(), (c ? a = tmp : b = tmp);
```

Reply:
The more I stare, the more I think `checkAddressable` and `toLvalue` just do the same thing.  I'll have a look into possibly merging them later.

Reply:
Anyway, I'm getting distracted. #21678.